src/Nethermind/Nethermind.Consensus/Processing/BranchProcessor.cs

```diff
@@ -98,10 +101,9 @@ public Block[] Process(BlockHeader? baseBlock, IReadOnlyList<Block> suggestedBlo
                 }
                 // If prewarmCancellation is not null it means we are in first iteration of loop
                 // and started prewarming at method entry, so don't start it again
-                if (prewarmCancellation is null)
-                {
-                    (prewarmCancellation, preWarmTask) = PreWarmTransactions(suggestedBlock, preBlockBaseBlock, spec);
-                }
+                backgroundCancellation ??= new CancellationTokenSource();
+                preWarmTask ??= PreWarmTransactions(suggestedBlock, preBlockBaseBlock, spec, backgroundCancellation.Token);
+                prefetchBlockhash ??= blockhashProvider.Prefetch(suggestedBlock.Header, backgroundCancellation.Token);
```

Comment:
Probably need to make sure this finished to prevent prefetch of early blockhash to get applied when executing later blockc.

Reply:
Good catch, decided not to await the task to keep the BranchProcessor as fast as possible, but added synchronization in the Prefetch itself.